cmd/mrCheckout.go

```diff
@@ -36,18 +43,53 @@ var checkoutCmd = &cobra.Command{
 			fmt.Printf("MR #%d not found\n", mrID)
 			return
 		}
-		// https://docs.gitlab.com/ee/user/project/merge_requests/#checkout-merge-requests-locally
-		if branch == "" {
-			branch = mrs[0].SourceBranch
+
+		mr := mrs[0]
+		// If the config does not specify a branch, use the mr source branch name
+		if mrCheckoutCfg.branch == "" {
+			mrCheckoutCfg.branch = mr.SourceBranch
 		}
-		mr := fmt.Sprintf("refs/merge-requests/%d/head", mrID)
-		gitf := git.New("fetch", forkedFromRemote, fmt.Sprintf("%s:%s", mr, branch))
+		// By default, fetch to configured branch
+		fetchToRef := mrCheckoutCfg.branch
+
+		// If track, make sure we have a remote for the mr author and then set
+		// the fetchToRef to the mr author/sourceBranch
+		if mrCheckoutCfg.track {
+			// Check if remote already exists
+			if _, err := gitconfig.Local("remote." + mr.Author.Username + ".url"); err != nil {
+				// Find and create remote
+				mrProject, err := lab.GetProject(mr.ProjectID)
+				if err != nil {
+					log.Fatal(err)
+				}
+				err = git.RemoteAdd(mr.Author.Username, mrProject.SSHURLToRepo, ".")
+				if err != nil {
+					log.Fatal(err)
+				}
+			}
+			fetchToRef = fmt.Sprintf("refs/remotes/%s/%s", mr.Author.Username, mr.SourceBranch)
+		}
+
+		// https://docs.gitlab.com/ee/user/project/merge_requests/#checkout-merge-requests-locally
+		mrRef := fmt.Sprintf("refs/merge-requests/%d/head", mrID)
+		gitf := git.New("fetch", forkedFromRemote, fmt.Sprintf("%s:%s", mrRef, fetchToRef))
 		err = gitf.Run()
 		if err != nil {
 			log.Fatal(err)
 		}
 
-		gitc := git.New("checkout", branch)
+		if mrCheckoutCfg.track {
+			// Create configured branch with tracking from fetchToRef
+			// git branch --flags <branchname> [<start-point>]
+			gitb := git.New("branch", "--track", mrCheckoutCfg.branch, fetchToRef)
+			err = gitb.Run()
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+
+		// Check out branch
+		gitc := git.New("checkout", mrCheckoutCfg.branch)
```

Comment:
Kind of a bad pattern on my part is looks like. These can really just be combined like
```
err := git.New("checkout", mrCheckoutCfg.branch).Run()
```

Reply:
Ooooooooooooooooooooooooooooo. I like that :+1: